### ISSUE TITLE
Ping/RTT ?

### ISSUE THREAD

**Issue by RaidmaxSpace** [paulov-t/SIT.Core#25]
Is it normal that when the game starts we only have 2-5 ping/RTT but the more we play in the game the more it goes up in ping/RTT and it gets serious lag? we play in lan

l'hote perd pas de fps mais lag quand le ping commence a dépassé les 50ping et les RTT aussi. 

Le client perd des fps et lag aussi quand le ping dépasse les 50ping et les RTT aussi.

**Comment by RaidmaxSpace** [comment]
Okay, I understand.

You do great work, it's amazing!

but i noticed that in the log the ia respanw constantly we can't block the respanw? or limited to 5 in a map max that it stays light?

**Comment by RaidmaxSpace** [comment]
Yes we noticed that the military base was unplayable but the Wood map was much better 

I must say that you all do a good job :)

**Comment by RaidmaxSpace** [comment]
I have another question why I never see the client character move? But him the customer sees me move without problem

is there a rule setting for that?

**Comment by RaidmaxSpace** [comment]
> Merci! ❤️
> 
> Oui, je pense que limiter l'IA sera le seul moyen d'éviter ce problème. C'est pourquoi Factory est la meilleure carte en ce moment. (et je déteste Factory haha)
> 
> J'ai hâte de voir Solarint/Props/DrakiaXYZ travailler sur l'IA, ce qui signifie que nous pourrions réduire la quantité d'IA, mais ils seront BEAUCOUP meilleurs pour naviguer sur la carte.

Oops sorry for the post notification,

for the moment I'm trying to set the ia "number of spawns and reappearance" parameter of the factory map for the other maps to keep it playable, until the real developer manage it ;) 

> Thanks! ❤️
> 
> Yes, I think that limiting AI will be the only way to avoid this issue. This is why Factory is the best map right now. (and I hate Factory haha)
> 
> I am looking forward to seeing Solarint/Props/DrakiaXYZ work on AI which means we could reduce the AI amount but they will be MUCH better at navigating the map.

///////////////////////////////////

Oops sorry for the post notification,

for the moment I'm trying to set the ia "number of spawns and reappearance" parameter of the factory map for the other maps to keep it playable, until the real developer manage it ;)

**Comment by pitAlex** [comment]
I think what you should do is rely on web sockets. Node is very good at handling that. You can send JSON with them, there even is a library for that (https://github.com/elpheria/rpc-websockets).  Now this would require the server machine to have node installed, and I think its work it. Your Aki mod could fire up this server, not use the Aki one because I don't see a way there for you to be able to utilize clusters. That's what you need. And so you spawn the child process and they will be able to handle requests much faster and in parallel. And then this server can communicate internally with Aki, for things that are not player-to-player synchronization.  And then on the client, I would see making use of Reactive X to have observables so you can send data to this server and back in an asynchronous way. 
Just my thoughts, I know node, I don't know C#.

**Comment by RaidmaxSpace** [comment]
> I think what you should do is rely on web sockets. Node is very good at handling that. You can send JSON with them, there even is a library for that (https://github.com/elpheria/rpc-websockets). Now this would require the server machine to have node installed, and I think its work it. Your Aki mod could fire up this server, not use the Aki one because I don't see a way there for you to be able to utilize clusters. That's what you need. And so you spawn the child process and they will be able to handle requests much faster and in parallel. And then this server can communicate internally with Aki, for things that are not player-to-player synchronization. And then on the client, I would see making use of Reactive X to have observables so you can send data to this server and back in an asynchronous way. Just my thoughts, I know node, I don't know C#.

Hello,

I'm not as strong as you, hahaha

I will do my best, hahaha

Thank you for your answer.

**Comment by RaidmaxSpace** [comment]
THEN the ping/RTT remains stable, but still this character problem that does not move between the client and the host in both directions

**Comment by JediKings1937** [comment]
@paulov-t Did not fix the issue of shooting causing jamming.. What should we do? God!
